agent/agentbootstrap/bootstrap.go

```diff
@@ -201,22 +204,49 @@ func (b *AgentBootstrap) Initialize(ctx stdcontext.Context) (_ *state.Controller
 
 	stateParams := b.stateInitializationParams
 
+	dqliteBootstrapOps := []database.BootstrapOpt{
+		ccbootstrap.InsertInitialControllerConfig(stateParams.ControllerConfig),
+		cloudbootstrap.InsertCloud(stateParams.ControllerCloud),
+	}
+
 	// Add the controller model cloud and credential to the database.
 	cloudCred, cloudCredTag, err := b.getCloudCredential()
 	if err != nil {
 		return nil, errors.Annotate(err, "getting cloud credentials from args")
 	}
-	ctrlCloud := stateParams.ControllerCloud
-	ctrlCloud.IsControllerCloud = true
+	if cloudCredTag.Id() != "" {
+		dqliteBootstrapOps = append(dqliteBootstrapOps,
+			credbootstrap.InsertCredential(cloudCredTag, cloudCred),
+		)
+	}
+
+	controllerModelType := modeldomain.TypeIAAS
+	if cloud.CloudIsCAAS(stateParams.ControllerCloud) {
+		controllerModelType = modeldomain.TypeCAAS
+	}
+
+	controllerUUID := modeldomain.UUID(
+		stateParams.ControllerModelConfig.UUID(),
+	)
+	controllerModelArgs := modeldomain.ModelCreationArgs{
+		Name:        stateParams.ControllerModelConfig.Name(),
+		Owner:       b.adminUser.Name(),
+		Cloud:       stateParams.ControllerCloud.Name,
+		CloudRegion: stateParams.ControllerCloudRegion,
+		Credential:  credential.IdFromTag(cloudCredTag),
+		Type:        controllerModelType,
+	}
+	dqliteBootstrapOps = append(
+		dqliteBootstrapOps,
+		modelbootstrap.CreateModel(controllerUUID, controllerModelArgs),
+	)
 
 	if err := b.bootstrapDqlite(
-		ctx,
+		stdcontext.TODO(),
```

Comment:
Why replace the passed in context with a TODO?

Reply:
was a bad rebase I just missed it in the dirty conflicts.